src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -597,14 +598,22 @@ ColumnsDescription InterpreterCreateQuery::getColumnsDescription(
                     setAlias(col_decl.default_expression->clone(), tmp_column_name));
             }
             else
+            {
+                has_columns_with_default_without_type = true;
                 default_expr_list->children.emplace_back(setAlias(col_decl.default_expression->clone(), col_decl.name));
+            }
         }
     }
 
     Block defaults_sample_block;
-    /// set missing types and wrap default_expression's in a conversion-function if necessary
-    if (!default_expr_list->children.empty())
+    /// Set missing types and wrap default_expression's in a conversion-function if necessary.
+    /// We try to avoid that validation while restoring from a backup because it might be slow or troublesome
+    /// (for example, a default expression can contain dictGet() and that dictionary can access remote servers or
+    /// require different users to authenticate).
+    if (!default_expr_list->children.empty() && (has_columns_with_default_without_type || (mode <= LoadingStrictnessLevel::CREATE)))
```

Comment:
AFAIU we now disable it for attach also, is that okay?
A new version could break the default expression.

Reply:
I know. From the first glance it looks ok not to evaluate default expressions during `ATTACH`, but let's try asking @alesapin because he added this line a long ago.

Reply:
The default expressions are evaluated during `CREATE TABLE` and it seems unnecessary to evaluate them during `ATTACH TABLE` again.

Reply:
okay, we would actually get better behavior I think.
CH instance will be able to start but reject inserts with broken defaults if we break it somehow in new version.

Just wanted to raise the issue and hear the opinion, I'm fine either way.